google-http-client/src/main/java/com/google/api/client/http/HttpRequest.java

```diff
@@ -1002,8 +1002,6 @@ public HttpResponse execute() throws IOException {
 
       // execute
       lowLevelHttpRequest.setTimeout(connectTimeout, readTimeout);
-      // switch tracing scope to current span
-      Scope ws = tracer.withSpan(span);
```

Comment:
This line switches the context to the newly created span.
We always create a new span in case user does not have one.